### PR TITLE
Added note about copying elements during conversion

### PR DESCRIPTION
**Pull request by magnolia-k**
https://github.com/scala/bug/issues/11872

In addition, the method introduction was modified to the latest method list.

### CODE REVIEW

File: _overviews/collections-2.13/trait-iterable.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Collection classes that implement `Iterable` just need to define this method; al
...
 * **Map** operations `map`, `flatMap`, and `collect`, which produce a new collection by applying some function to collection elements.
-* **Conversions** `toArray`, `toList`, `toIterable`, `toSeq`, `toIndexedSeq`, `toStream`, `toSet`, `toMap`, which turn an `Iterable` collection into something more specific. All these conversions return their receiver argument unchanged if the run-time type of the collection already matches the demanded collection type. For instance, applying `toList` to a list will yield the list itself.
+* **Conversions** `to`, `toList`, `toVector`, `toMap`, `toSet`, `toSeq`, `toIndexedSeq`, `toBuffer`, `toArray` which turn an `Iterable` collection into something more specific. If the destination is a mutable collection(to(collection.mutable.X), to(collection.X), toArray, toBuffer), a new collection is created by copying the original elements. All these conversions return their receiver argument unchanged if the run-time type of the collection already matches the demanded collection type. For instance, applying `toList` to a list will yield the list itself.
```
Comment (SethTisue) [inline _overviews/collections-2.13/trait-iterable.md]: looks like some missing backquotes here, to make `to(collection.mutable.X)` and so forth render in monospace font

Comment (magnolia-k) [reply _overviews/collections-2.13/trait-iterable.md]: thanks!
fixed it.

**Comment by SethTisue** [comment]
`If the destination is a mutable collection(`to(collection.mutable.X)`, `to(collection.X)`

I'm wondering about the presence of `to(collection.X)` here. Those collections aren't necessarily mutable. Is it actually true that copying always occurs? Maybe it is, but it would surprise me.

And regardless, it doesn't seem correct to me for the phrasing to imply that `collection.X` is mutable, since it might or might not be.

**Comment by SethTisue** [comment]
interested in getting this across the finish line?

**Comment by SethTisue** [comment]
@magnolia-k interested in getting this across the finish line?

**Comment by magnolia-k** [comment]
@SethTisue 

I'm sorry it took me so long to reply.

It's true, as you point out, not all of the to(collection.X) generates a copy.

**Comment by SethTisue** [comment]
thank you!

**Pull request merged**